homeassistant/components/watts/coordinator.py

```diff
@@ -0,0 +1,99 @@
+"""Data coordinator for Watts Vision integration."""
+
+from __future__ import annotations
+
+from datetime import timedelta
+import logging
+
+from visionpluspython.client import WattsVisionClient
+from visionpluspython.models import Device
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.update_coordinator import DataUpdateCoordinator, UpdateFailed
+
+from .const import DOMAIN, UPDATE_INTERVAL
+
+_LOGGER = logging.getLogger(__name__)
+
+
+class WattsVisionCoordinator(DataUpdateCoordinator):
```

Comment:
Okay so for the coordinator, currently you have 1 coordinator for all the states. The thing is that users can use `homeassistant.update_entity` to manually trigger an update. Currently this would update all devices. 

I am wondering if it would make sense to just have one coordinator per device, this way we don't have to worry about updating only device in a dict and whatnot, we just ask the coordinator to update and it will update a single device. Would that make sense?